src/SFA.DAS.FAT.Web/Controllers/CoursesController.cs

```diff
@@ -23,311 +23,308 @@
             }
 
-            return new LocationCookieItem
-            {
-                Name = location
-            };
+            var removedProviderFromShortlist =
+                string.IsNullOrEmpty(removed) ? "" : HttpUtility.HtmlDecode(GetEncodedProviderName(removed));
+            var addedProviderToShortlist =
+                string.IsNullOrEmpty(added) ? "" : HttpUtility.HtmlDecode(GetEncodedProviderName(added));
+
+            viewModel.BannerUpdateMessage = GetProvidersBannerUpdateMessage(removedProviderFromShortlist, addedProviderToShortlist);
+
+            viewModel.HelpFindingCourseUrl = BuildHelpFindingCourseUrl(viewModel.Course.Id, EntryPoint.ProviderDetail);
+
+            // var currentDate = _dateTimeService.GetDateTime();
+            //
+            // if (currentDate.Month >= 3 && currentDate.Day > 30)
+            // {
+            //     viewModel.AchievementRateFrom = currentDate.Year - 2;
+            // }
+            // else
+            // {
+            //     viewModel.AchievementRateFrom = currentDate.Year - 3;
+            // }
+            viewModel.AchievementRateFrom = 2022;
+            return View(viewModel);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, e.Message);
+            return RedirectToRoute(RouteNames.Error500);
         }
+    }
 
-        private void UpdateLocationCookie(LocationCookieItem location)
+    private LocationCookieItem CheckLocation(string location)
+    {
+        if (location == "-1")
         {
-            if (!string.IsNullOrEmpty(location.Name) && location.Lat != 0 && location.Lon != 0)
-            {
-                _locationCookieStorageService.Update(Constants.LocationCookieName, location, 2);
-            }
+            _locationCookieStorageService.Delete(Constants.LocationCookieName);
+            return null;
+        }
+        if (string.IsNullOrEmpty(location))
+        {
+            return _locationCookieStorageService.Get(Constants.LocationCookieName);
         }
 
-        private string GetEncodedProviderName(string encodedName)
+        return new LocationCookieItem
         {
-            try
-            {
-                var base64EncodedBytes = WebEncoders.Base64UrlDecode(encodedName);
-                return System.Text.Encoding.UTF8.GetString(_shortlistDataProtector.Unprotect(base64EncodedBytes));
-            }
-            catch (FormatException e)
-            {
-                _logger.LogInformation(e, "Unable to decode provider name from request");
-            }
-            catch (CryptographicException e)
-            {
-                _logger.LogInformation(e, "Unable to decode provider name from request");
-            }
+            Name = location
+        };
+    }
 
-            return string.Empty;
+    private void UpdateLocationCookie(LocationCookieItem location)
+    {
+        if (!string.IsNullOrEmpty(location.Name) && location.Lat != 0 && location.Lon != 0)

```

Comment:
lon = 0 is a valid value.  I think this should be more
if (!string.IsNullOrEmpty(location.Name) && !(location.Lat == 0 && location.Lon == 0)

Reply:
Location cookie is going away anyway